src/server/scripts/EasternKingdoms/zone_silverpine_forest.cpp

```diff
@@ -1,139 +1,2175 @@
- * with this program. If not, see <http://www.gnu.org/licenses/>.
- */
+/*
+* This file is part of the TrinityCore Project. See AUTHORS file for Copyright information
+*
+* This program is free software; you can redistribute it and/or modify it
+* under the terms of the GNU General Public License as published by the
+* Free Software Foundation; either version 2 of the License, or (at your
+* option) any later version.
+*
+* This program is distributed in the hope that it will be useful, but WITHOUT
+* ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+* FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+* more details.
+*
+* You should have received a copy of the GNU General Public License along
+* with this program. If not, see <http://www.gnu.org/licenses/>.
+*/
 
 /* ScriptData
 SDName: Silverpine_Forest
-SD%Complete: 100
-SDComment: Quest support: 435
+SD%Complete: 20 %
+SDComment:
 SDCategory: Silverpine Forest
 EndScriptData */
 
-/* ContentData
-npc_deathstalker_erland
-EndContentData */
+#include "CombatAI.h"
+#include "GameObject.h"
+#include "GameObjectAI.h"
+#include "GridNotifiers.h"
+#include "MotionMaster.h"
+#include "ObjectAccessor.h"
+#include "ObjectMgr.h"
+#include "Object.h"
+#include "Player.h"
+#include "ScriptMgr.h"
+#include "ScriptedCreature.h"
+#include "ScriptedGossip.h"
+#include "ScriptedCreature.h"
+#include "TemporarySummon.h"
+#include "Unit.h"
+#include "Vehicle.h"
+
+// Worgen Renegade - 44793
+struct npc_worgen_renegade_44793 : public ScriptedAI
+{
+    npc_worgen_renegade_44793(Creature* creature) : ScriptedAI(creature) { }
+
+    enum eNPC
+    {
+        NPC_FORSAKEN_TROOPER1 = 44791,
+        NPC_FORSAKEN_TROOPER2 = 44792,
+
+        SPELL_HEARTSTRIKE = 84182,
+        SPELL_DARKENED = 73307,
+        SPELL_FLURRY_OF_CLAWS_CHANNEL = 80365,
+        SPELL_FLURRY_OF_CLAWS = 80367,
+
+        EVENT_CHECK_SHOWFIGHT = 101,
+        EVENT_SPELL_QUEUE = 102,
+        EVENT_SPELL_DAMAGE = 103,
+
+        MOVE_TO_HOMEPOSITION,
+    };
+
+    EventMap m_events;
+    bool m_isShowFight;
+
+    void Reset() override
+    {
+        m_events.Reset();
+
+        me->RemoveAura(SPELL_DARKENED);
+
+        m_isShowFight = true;
+
+        m_events.ScheduleEvent(EVENT_CHECK_SHOWFIGHT, 1s);
+    }
+
+    void SpellHit(Unit* caster, SpellInfo const* spell) override
+    {
+        if (!caster)
+            return;
+
+        if (spell->Id == SPELL_HEARTSTRIKE)
+            me->AddAura(SPELL_DARKENED, me);
+    }
+
+    void MovementInform(uint32 type, uint32 pointId) override
+    {
+        if (type == POINT_MOTION_TYPE && pointId == MOVE_TO_HOMEPOSITION)
+            m_events.ScheduleEvent(EVENT_CHECK_SHOWFIGHT, 25ms);
+    }
+
+    void JustEngagedWith(Unit* /*who*/) override
+    {
+        m_events.ScheduleEvent(EVENT_SPELL_QUEUE, 3s);
+    }
+
+    void UpdateAI(uint32 diff) override
+    {
+        m_events.Update(diff);
```

Comment:
private members of a class should use  _lowerCamelCase, m_lowerCameCase is outdated

Reply:
Fixed.